runtime/parser2/expression_test.go

```diff
@@ -313,6 +315,25 @@ func TestParseAdvancedExpression(t *testing.T) {
 		)
 	})
 
+	t.Run("test FatalError in setExprMetaLeftDenotation", func(t *testing.T) {
+
+		t.Parallel()
+
+		gauge := makeLimitingMemoryGauge()
+		gauge.Limit(common.MemoryKindIntegerExpression, 1)
+
+		var panicMsg interface{}
+		(func() {
+			defer func() {
+				panicMsg = recover()
+			}()
+
+			ParseExpression("1 < 2 > 3", gauge)
+		})()
+
+		require.IsType(t, common.FatalError{}, panicMsg)
```

Comment:
Maybe also assert the inner error is a memory limit error (and not potentially some other error in the future)